### ISSUE TITLE
Lens coat thickness

### ISSUE THREAD

**Issue by Nickelium** [greje656/PhysicallyBasedLensFlare#5]
Hi,

I`m wondering how you found/compute/chose the lens coat thickness (PatentFormat.c == LensInterface.d1), which you said correspond to the wavelength the lens is optimized for (".. In the current implementation each lens coating specifies a wavelength the coating should be optimized ..") ? I couldnt find any patent describing the coating, so I thought it might be hand picked but probably not as it correponds to https://github.com/AngularSpectrumMTD/Physically-Based-Realtime-Lens-Flare.

**Comment by AngularSpectrumMTD** [comment]
I cited the lens params(nikon and angenieux) from this repository.
So, my code has same param correspond to this repository.

Dear Jean-Philippe Grenier (greje656), owner of this repository.

I apologize for the unauthorized use of your analyzed patent data.
After simulated flare by using params of canon, I wanted to try other lens.
Then, I searched lens params on web, I found the parsed param of Nikon and Angenieux on this repository.

I'm sorry for using parsed params without your permission.
I always turn my hobby repository for validation to private mode, but I forgot to turn my repository private.
It has now been removed from the sights of others.
I am sorry for the discomfort I caused you.

**Comment by Polyrhythm** [comment]
@Nickelium  I'm not sure the answer to this but it is perhaps just random selection based on trying to make the flare ghosts look more colorful.  The original paper even says that AR coatings are a trade secret so it's highly unlikely they'd be listed in the patent. I guess if you had the lens on hand you could estimate the wavelength of each element by illuminating the lens body with a white light and looking at the reflections.

He does seem to include a consistent "440" for aperture elements, not sure the deal there.

By the way, I am still trying to figure out out how W and H are calculated as well, do you know that?
The patent format used here is R, D, N, isAp, W, H, C.

**Comment by Nickelium** [comment]
@Polyrhythm Yeah that's plausible explanation. If that's the case then it's a bit sad.

Good observation! Though the coat thickness for the aperture is irrelevant for the color, as the lights ray just goes through the aperture (without reflection/refraction). Only lens element where rays reflect off, contribute to the color, I believe. The aperture is technically an opening and thus shouldn`t have a lens on it, thus no coat thickness as result.

I didn't find either but haven't look in much as I`m rather trying to implement the second paper (which disregards the heights and widths except from the aperture and the entrance lens). But let me know if you find something.
According to https://placeholderart.wordpress.com/2015/01/19/implementation-notes-physically-based-lens-flares/,
"  .. some information missing when reading the patent, namely the heights of the various elements .. I just took a guess at the various element heights .. "

Have you looked at the second paper aswell?
